src/Sarif.Viewer.VisualStudio/LogProcessedEventArgs.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Sarif.Viewer
+{
+    // Provides data for the event handler invoked when the ErrorListService finishes
+    // processing a SARIF log.
+    internal class LogProcessedEventArgs
+    {
+        internal LogProcessedEventArgs(ExceptionalConditions exceptionalConditions)
+        {
+            ExceptionalConditions = exceptionalConditions;
+        }
+
+        // Gets any exceptional conditions (for example, an error-level tool execution
+        // failure) that occurred during the processing of the log file.
+        internal ExceptionalConditions ExceptionalConditions { get; }
```

Comment:
>internal [](start = 8, length = 8)

The coding style is really up to you, but double internal is something we typically don't do. The class can be internal, and this method is public. (in my opinioin)

Great article on this

[https://ericlippert.com/2014/09/15/internal-or-public/](https://ericlippert.com/2014/09/15/internal-or-public/)

And the "accepted answer" on stack-overflow
[https://stackoverflow.com/questions/9302236/why-use-a-public-method-in-an-internal-class#:~:text=If%20the%20class%20is%20internal%2C%20it%20doesn%27t%20matter,as%20part%20of%20the%20description%20of%20the%20method](https://stackoverflow.com/questions/9302236/why-use-a-public-method-in-an-internal-class#:~:text=If%20the%20class%20is%20internal%2C%20it%20doesn%27t%20matter,as%20part%20of%20the%20description%20of%20the%20method).

Reply:
Haven't seen this issue come up in a while! :) I agree with Eric's point: internal classes should organize their methods in a way that clarifies what is 'public' surface area (given the constraint that the containing scope is restricted to the assembly and any 'internals visible to' attribute. The way I frame this is imagine that the class itself is appropriately made public (i.e., cross-assembly) in the future: the accessibility modifiers should still make sense. Note that this *does* lead to the possibility of internal/internal classification, for things like test helpers that would never be a good candidate for cross-assembly access.

---
In reply to: [527123309](https://github.com/microsoft/sarif-visualstudio-extension/pull/278#discussion_r527123309) [](ancestors = 527123309)

Reply:
Ok, `internal`/`public` makes sense. Note that either (1) When you make the class public, you have to write the XML doc comments then, or (2) You write XML doc comment up front on anything marked `public`. I assume @mikefan@microsoft.com wants #1.

---
In reply to: [527127708](https://github.com/microsoft/sarif-visualstudio-extension/pull/278#discussion_r527127708) [](ancestors = 527127708,527123309)

Reply:
i think that only applies if the class is public...but i might be wrong..

Reply:
You are right, the _requirement_ for XML comments is only if the class is public. And as long as the team understands that the price of making a class public includes writing the comments at the time the decision is made, then all is well.

---
In reply to: [527130151](https://github.com/microsoft/sarif-visualstudio-extension/pull/278#discussion_r527130151) [](ancestors = 527130151)